### ISSUE TITLE
Coverage rate is incorrect when some code fragments are ignored

### ISSUE THREAD

**Issue by hanhongfang** [gotwarlost/istanbul#218]
Hi dev,

I'm using /_istanbul ignore next_/ to ignore some code fragments. For example, I totally have 1000 lines, and I ignore 900 lines. I have unit tests which covered 50 lines of the left 100 lines.  Currently istanbul reports coverage rate as 950 / 1000 = 95%, which takes the 900 lines being ignored as tested. But I think the coverage rate should focus on the left 100 lines only. Within the left 100 lines, only 50 lines are tested, thus the coverage rate should be 50%. [In my case, 900 lines are library, which is built into a combined single javascript file with my code, but testing to the library is not my purpose.]

Is there any configuration that I can use to make istanbul calculate the rate as I expected?

Best regards,
Janet Han

**Comment by technicallyfeasible** [comment]
Agreed, I have some boilerplate code that is autogenerated by a babel transpile and is obviously not my code. When I ignore it I would expect this not to be set to "1" or even more if it is actually hit but instead be removed from the baseline coverage and only the rest of the code count towards testable lines.
Here's an example image:
![istanbul_coverage](https://cloud.githubusercontent.com/assets/5169186/9088893/5b31ad96-3b8a-11e5-8d6e-b7d708ba71a0.png)

**Comment by destroyerofbuilds** [comment]
I have a similar scenario to @technicallyfeasible

**Comment by hopeson** [comment]
+1

**Comment by technicallyfeasible** [comment]
I ended up creating a patch that I include before my tests:

https://gist.github.com/technicallyfeasible/5f066d323ddf608e3cac

This modifies the computeBranchTotals and computeSimpleTotals functions to report only stats which were not excluded.

Simply do require('patch_istanbul') before getting the test run totals or generating the report.
